src/main/drivers/display_ug2864hsweg01.c

```diff
@@ -24,16 +24,31 @@
 
 #include "display_ug2864hsweg01.h"
 
+#include "config/parameter_group.h"
+#include "config/parameter_group_ids.h"
+
 #ifdef USE_I2C_OLED_DISPLAY
 
 #if !defined(OLED_I2C_INSTANCE)
 #if defined(I2C_DEVICE)
 #define OLED_I2C_INSTANCE I2C_DEVICE
 #else
-#define OLED_I2C_INSTANCE I2C_NONE
+#define OLED_I2C_INSTANCE I2CINVALID
 #endif
 #endif
 
+#define OLED_address   0x3C     // OLED at address 0x3C in 7bit
+
+PG_REGISTER_WITH_RESET_TEMPLATE(oledDisplayConfig_t, oledDisplayConfig, PG_OLED_DISPLAY_CONFIG, 0);
+
+PG_RESET_TEMPLATE(oledDisplayConfig_t, oledDisplayConfig,
+    .bus = OLED_I2C_INSTANCE,
+    .address = OLED_address,
+);
+
+static I2CDevice i2cBus = I2CINVALID; // XXX Protect against direct call to i2cWrite and i2cRead
+static uint8_t i2cAddress;
```

Comment:
Device driver code should be re-entrant to allow multiple instances of the device. So we should not have `static` I2CDevice and address. Rather we should define (in `drivers/bus.h`)

typedef union busDevice_u {
    struct deviceSpi_s {
        IO_t csnPin;
    } spi;
    struct deviceI2C_s {
       I2CDevice device;
       uint8_t address;
    } i2c;
} busDevice_t;

Then define `i2cWrite` as `bool i2cWrite(busDevice_t *bus, uint8_t addr_, uint8_t reg, uint8_t data);`, `static bool i2c_OLED_send_cmd(uint8_t command)` becomes `static bool i2c_OLED_send_cmd(busDevice_t *bus, uint8_t command)` etc etc.